app/controllers/grades_controller.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class GradesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_grade, only: [:create, :destroy]
+
+  def create
+    @grade = @grade.present? ? @grade : Grade.new(assignment_id: grade_params[:assignment_id])
```

Comment:
If grade is already present, then it should got to update action, not create

Reply:
I chose to write one route for creating and updating grades so that in the front end it wasn't necessary to know whether the grade for the project had been created or not to decide which route to call. Also, I felt it was easier to check if the grade was created in the back end as compared to the front end. 😅

Reply:
But logic will be messy in the future, We have to keep the code maintainable in long term

Reply:
I feel that the front end logic might get messy for checking in between post, put, delete requests for grades and changing them as and when we create/update/delete grades. Is there an easy way to handle it on the front end?

Reply:
I think what @ankitkataria has done here is fine. Ideally, it should be separate but it is not a very big deal.